server/pulp/server/webservices/views/repositories.py

```diff
@@ -0,0 +1,1214 @@
+    def get(self, request, repo_id, distributor_id):
+        """
+        :param request: WSGI request object
+        :type  request: django.core.handlers.wsgi.WSGIRequest
+        :param repo_id: The id of the repository the distributor is associated with
+        :type  repo_id: str
+        :param repo_id: The id of the requested distributor
+        :type  repo_id: str
+
+        :return: Response containing a dict of the requested distributor
+        :rtype : django.http.HttpResponse
+        """
+
+        distributor_manager = manager_factory.repo_distributor_manager()
+        distributor = distributor_manager.get_distributor(repo_id, distributor_id)
+        return generate_json_response_with_pulp_encoder(distributor)
+
+    @auth_required(authorization.DELETE)
+    def delete(self, request, repo_id, distributor_id):
+        """
+        Disassociate the requested distributor.
+
+        :param request: WSGI request object
+        :type  request: django.core.handlers.wsgi.WSGIRequest
+        :param repo_id: The id of the repository the to disassociate from
+        :type  repo_id: str
+        :param repo_id: The id of the distributor to disassociate
+        :type  repo_id: str
+
+        :raises pulp_exceptions.OperationPostponed: dispatch a task
+        """
+
+        # validate resources
+        manager = manager_factory.repo_distributor_manager()
+        manager.get_distributor(repo_id, distributor_id)
+
+        task_tags = [
+            tags.resource_tag(tags.RESOURCE_REPOSITORY_TYPE, repo_id),
+            tags.resource_tag(tags.RESOURCE_REPOSITORY_DISTRIBUTOR_TYPE, distributor_id),
+            tags.action_tag('remove_distributor')
+        ]
+        async_result = repo_tasks.distributor_delete.apply_async_with_reservation(
+            tags.RESOURCE_REPOSITORY_TYPE, repo_id, [repo_id, distributor_id],
+            tags=task_tags)
+        raise pulp_exceptions.OperationPostponed(async_result)
+
+    @auth_required(authorization.UPDATE)
+    @json_body_required
+    def put(self, request, repo_id, distributor_id):
+        """
+        Used to update a repo distributor instance.
+
+        The expected parameters are 'distributor_config', which is a dictionary containing
+        configuration values accepted by the distributor type, and 'delta', which is a dictionary
+        containing other configuration values for the distributor (like the auto_publish flag,
+        for example). Currently, the only supported key in the delta is 'auto_publish', which
+        should have a boolean value.
+
+        :param request: WSGI request object
+        :type  request: django.core.handlers.wsgi.WSGIRequest
+        :param repo_id: id of the repository the distributor is associated with
+        :type  repo_id: str
+        :param distributor_id: id of the distributor instance to update.
+        :type  distributor_id: str
+
+        :raises pulp_exceptions.MissingValue: dispatch a task
```

Comment:
s/:raises pulp_exceptions.MissingValue: dispatch a task/raises pulp_exceptions.MissingValue: if distributor_config is none